### DISCUSSION TITLE
nvim-dap doesn't work right with codelldb

### DISCUSSION THREAD

**Discussion by imindMan** [mfussenegger/nvim-dap#1146] [Q&A]
I have started using Neovim now, and I really have some troubles with my Neovim configuration, specifically nvim-dap itself. I have tried to configure it to work for my Rust workflow and it just doesn't work for me then.

My configuration

```lua
local dap = require("dap")
dap.adapters.codelldb = {
      type = "server",
      port = "${port}",
      executable = {
          command = "$HOME/.local/share/nvim/mason/bin/codelldb" -- I installed codelldb through mason.nvim
          args = {"--port", "${port}"},

      },
}

dap.configurations.rust = {
      {
           name = "Launch",
           type = "codelldb",
           request = "launch",
           program = function()
                  return vim.fn.input("Path to executable: ", vim.fn.getcwd() .. "/target/debug/", "file")
           end,
           cwd = "${workspaceFolder}",
           stopOnEntry = false,
      },
}

```

It looks like it should work for me. And indeed, it works just fine until I open a `:DapToggleBreakpoint` command to specify a breakpoint, make a new session with `:DapContinue`. As you expected, it will show the dialog to let me input my executable file. So I did that, and instead of letting me step manually through the program to debug, it just runs the executable file and ends the session.

Note that I haven't touched the rustaceanvim plugin to configure my dap yet, I just installed and configured default stuff to let my lsp working. 

Please let me know if you have any ideas

**Comment by lewissettercaci** [comment]
Do you see `debug_info` in the output of `file BINARY_NAME `? 

See https://github.com/mfussenegger/nvim-dap/wiki/Troubleshooting

**Reply by mikesmithgh** [reply]
I had a similar issue and as @lewissettercaci said the steps in https://github.com/mfussenegger/nvim-dap/wiki/Debug-symbols-in-various-languages-and-build-systems worked for me.

When building with `gcc`, for example, make sure to pass `-g` to include debug symbols. e.g., `gcc -g hello.c`

**Reply by ghost** [reply]
I had also a similar issue, and I'm sure that my cpp executable include debug symboles but it just ignore all the breakpoints.
this probleme is just in cpp debuging when i test Rust with codelldb the debugger work, and when I test cppdbg (cpptools) as a debugger it work with the same cpp executable just codelldb that give me this issue

**Reply by lewissettercaci** [reply]
@gGameDevelop Can you paste the output of `file <executable>` for executable that fails to work with codelldb? If your breakpoint is in a dynamic library, does the include debug symbols?

**Reply by ghost** [reply]
No it's just one main.cpp with the std lib
